kalmanjax/priors.py

```diff
@@ -1092,12 +1092,17 @@ class SpatioTemporalMatern52(Prior):
         temporal lengthscale, lt
         spatial lengthscale, ls
     """
-    def __init__(self, variance=1.0, lengthscale_time=1.0, lengthscale_space=1.0, z=None, fixed_grid=False):
+    def __init__(self, variance=1.0, lengthscale_time=1.0, lengthscale_space=1.0, spatial_dims=1, z=None, fixed_grid=False):
         hyp = [variance, lengthscale_time, lengthscale_space]
         super().__init__(hyp=hyp)
+        self.spatial_dims = spatial_dims
         if z is None:
             z = np.linspace(-3., 3., num=15)
-        self.z = z.reshape(-1, 1)
+            if self.spatial_dims > 1:  # Quick, inelegant fix for |r| = 2
```

Comment:
This certainly not optimized nor generalized for >2 dims but it's a start... 🤔

Reply:
This looks good to me, but could we maybe raise a 'NotImplementedError' for the case where spatial_dims > 2? That way if someone tries to use more spatial dimensions they won't have to search for the bug manually like you had to.
Something like this would be good:

`raise NotImplementedError('initialisation of z with more than 2 dimensions not implemented, please provide a custom initialisation')`

Reply:
Good point, I'll add this in.